go/vt/vtctl/grpcvtctldserver/server.go

```diff
@@ -128,6 +129,30 @@ func (s *VtctldServer) ApplyRoutingRules(ctx context.Context, req *vtctldatapb.A
 	return resp, nil
 }
 
+// ApplyVSchema is part of the vtctlservicepb.VtctldServer interface.
+func (s *VtctldServer) ApplyVSchema(ctx context.Context, req *vtctldatapb.ApplyVSchemaRequest) (*vtctldatapb.ApplyVSchemaResponse, error) {
+	if _, err := s.ts.GetKeyspace(ctx, req.Keyspace); err != nil {
+		if topo.IsErrType(err, topo.NoNode) {
+			return nil, fmt.Errorf("keyspace(%s) doesn't exist, check if the keyspace is initialized", req.Keyspace)
```

Comment:
I would make this an `vterrors.Wrapf(vtrpc.FAILED_PRECONDITION, ...)`

Reply:
not totally sure what you meant here.
Wrap/Wrapf don't take an error code like failed precondition.
New and Errorf do take a code, but those are for making a new error, whereas here i already have an error.